### ISSUE TITLE
Crash when render SVG

### ISSUE THREAD

**Issue by Brooooooklyn** [nical/lyon#351]
https://resource.arkie.cn/resource/decoration/fc2768a0-7bd6-4464-9c47-59192584463e.svg

```bash
./target/debug/svg_render crash.svg
```
got: 
```bash
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:335:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
             at libstd/panicking.rs:463
   5: std::panicking::try::do_call
             at libstd/panicking.rs:350
   6: std::panicking::try::do_call
             at libstd/panicking.rs:328
   7: <&'a T as core::fmt::Debug>::fmt
             at libcore/panicking.rs:71
   8: <&'a T as core::fmt::Debug>::fmt
             at libcore/panicking.rs:51
   9: usvg::preproc::ungroup_switch::is_valid_child
             at /Users/travis/build/rust-lang/rust/src/libcore/macros.rs:20
  10: <f32 as num_traits::float::FloatConst>::FRAC_PI_4
             at ./geom/src/arc.rs:404
  11: <usvg::tree::attribute::PathSegment as core::clone::Clone>::clone
             at ./geom/src/arc.rs:151
  12: std::f64::<impl f64>::round
             at ./geom/src/arc.rs:349
  13: usvg::convert::path::convert_path
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/convert/path.rs:149
  14: usvg::convert::path::convert
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/convert/path.rs:38
  15: usvg::convert::convert_nodes
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/convert/mod.rs:278
  16: usvg::convert::convert_doc
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/convert/mod.rs:97
  17: usvg::preproc::resolve_pattern_children::resolve
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/lib.rs:121
  18: usvg::preproc::resolve_pattern_children::resolve
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/lib.rs:112
  19: usvg::parse_tree_from_file
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/usvg-0.1.1/src/lib.rs:153
  20: resvg::parse_rtree_from_file
             at /Users/longyinan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/resvg-0.2.0/src/lib.rs:214
  21: svg_render::main
             at examples/svg_render/src/main.rs:47
  22: std::rt::lang_start::{{closure}}
             at /Users/travis/build/rust-lang/rust/src/libstd/rt.rs:74
  23: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  24: panic_unwind::dwarf::eh::read_encoded_pointer
             at libpanic_unwind/lib.rs:105
  25: std::sys_common::bytestring::debug_fmt_bytestring
             at libstd/panicking.rs:289
             at libstd/panic.rs:374
             at libstd/rt.rs:58
  26: std::rt::lang_start
             at /Users/travis/build/rust-lang/rust/src/libstd/rt.rs:74
  27: svg_render::update_inputs
```

**Comment by zen3ger** [comment]
The line causing the issue is `geom/src/arc.rs:404`.
```rust
        let ctrl = l2.intersection(&l1).unwrap();
```
`intersection()` returns `None` here (`geom/src/line.rs:290`):
```rust
        if S::abs(det) <= S::EPSILON {
            // The lines are very close to parallel
            return None;
        }
```

This is what I found with GDB:
```
(gdb) b /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/lyon_geom-0.10.0/src/line.rs:283
Breakpoint 4 at 0x5555555d0b0e: file /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/lyon_geom-0.10.0/src/line.rs, line 283.
(gdb) r crash.svg 
Starting program: /home/zen3ger/Projects/lyon/target/debug/svg_render crash.svg
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

Breakpoint 4, <lyon_geom::line::Line<S>>::intersection (self=0x7fffffff9430, other=0x7fffffff9410) at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/lyon_geom-0.10.0/src/line.rs:283
283	            return None;
(gdb) p *self
$13 = Line<f32> = {point = TypedPoint2D<f32, euclid::UnknownUnit> = {x = 42.6800003, y = 101.920006, _unit = PhantomData<euclid::UnknownUnit>}, vector = TypedVector2D<f32, euclid::UnknownUnit> = {
    x = -3.8190683e-06, y = -0.0500000007, _unit = PhantomData<euclid::UnknownUnit>}}
(gdb) p *other
$14 = Line<f32> = {point = TypedPoint2D<f32, euclid::UnknownUnit> = {x = 42.6800003, y = 101.919998, _unit = PhantomData<euclid::UnknownUnit>}, vector = TypedVector2D<f32, euclid::UnknownUnit> = {
    x = 3.81032623e-06, y = -0.0500000007, _unit = PhantomData<euclid::UnknownUnit>}}
(gdb) n
296	    }
(gdb) n
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:335:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
0x00005555559c1eff in std::panicking::begin_panic_fmt () at libstd/panicking.rs:350
350	libstd/panicking.rs: No such file or directory.
```

**Comment by Brooooooklyn** [comment]
@RazrFalcon don't sure if this issue is related to resvg

**Comment by nical** [comment]
Thanks for filing this, I am travelling right now but I will get to this in a few days.

**Comment by nical** [comment]
I published `0.10.1` cherry-picking the fix for this bug.
